restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/CallManager.java

```diff
@@ -1727,11 +1816,15 @@ private void outboundToPstn(final CreateCall request, final ActorRef sender) thr
 
 
     private void outboundToSip(final CreateCall request, final ActorRef sender) throws ServletParseException {
+        final String uri = (request.getOutboundProxy()!= null && (!request.getOutboundProxy().isEmpty())) ? request.getOutboundProxy() : "";
```

Comment:
George, this is a specific requirement for URI-Rewrite. I think my understanding of SIP and usage of host versus proxies is still limited and incomplete (even after reading 3261) but the behaviour should be that the message going to `sip:number@host.com` will be routed to `outboundProxy` if it exists when we call the extension.
In a specific case the `outboundProxy` here would be NetworkIntelligence. They will then take the message we have forwarded which was rewritten to `sip:number@ni.com` and potentially modified with custom parameters and forward it to the original `sip:number@host.com` after they have processed the message.
https://docs.google.com/document/d/1RL9j1PHqHorhflroKBDyNs7ghH29t72IFMnh0JZGYTY/edit#heading=h.8s6nwkogs9wk

What is alternative to achieve this functionality aside from setting the host in the `SipURI`?
I do see a `message.getProxy` but there doesnt seem to be a `message.setProxy`

@deruelle @scottbarstow Jean, Scott, thoughts?